src/System.Private.CoreLib/src/System/Diagnostics/Eventing/RuntimeEventSourceHelper.Windows.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+
+
+namespace System.Diagnostics.Tracing
+{
+	internal sealed class RuntimeEventSourceHelper
+	{
+		internal static long prevProcUserTime = 0;
+		internal static long prevProcKernelTime = 0;
+		internal static long prevSystemUserTime = 0;
+		internal static long prevSystemKernelTime = 0;
+
+		internal static int GetCpuUsage()
+		{
+			// Returns the current process' CPU usage as a percentage
+
+			int cpuUsage;
+			
+			if (!Interop.Kernel32.GetProcessTimes(Interop.Kernel32.GetCurrentProcess(), out _, out _, out long procKernelTime, out long procUserTime))
+			{
+				throw new Exception("GetCpuUsage: Could not get process time.");
+			}
+
+			if (!Interop.Kernel32.GetSystemTimes(out _, out long systemUserTime, out long systemKernelTime))
+			{
+				throw new Exception("GetCpuUsage: Could not get system time.");
+			}
+
+			if (prevSystemUserTime == 0 && prevSystemKernelTime == 0) // These may be 0 when we report CPU usage for the first time, in which case we should just return 0. 
+			{
+				cpuUsage = 0;
+			}
+
+			else
+			{
+				long totalProcTime = (procUserTime - prevProcUserTime) + (procKernelTime - prevProcKernelTime);
+				long totalSystemTime = (systemUserTime - prevSystemUserTime) + (systemKernelTime - prevSystemKernelTime);
```

Comment:
Can `totalSystemTime` be zero here?

Reply:
I'd say it's very very unlikely if not impossible... This callback gets called only a single thread at most since there is only one instance of CPU counter, and the callback gets called at most once per second interval, whereas these times are measured in units of 100ns.

Reply:
ok